dbms/src/Storages/KVStore/Decode/DecodingStorageSchemaSnapshot.cpp

```diff
@@ -32,15 +121,24 @@ TMTPKType getTMTPKType(const IDataType & rhs)
     return TMTPKType::UNSPECIFIED;
 }
 
-Block createBlockSortByColumnID(DecodingStorageSchemaSnapshotConstPtr schema_snapshot)
+Block createBlockSortByColumnID(DecodingStorageSchemaSnapshotConstPtr schema_snapshot, bool has_version_column)
 {
     Block block;
-    for (auto iter = schema_snapshot->sorted_column_id_with_pos.begin();
-         iter != schema_snapshot->sorted_column_id_with_pos.end();
+    // # Safety
+    // Though `col_id_to_block_pos` lasks some fields in `col_id_to_def_pos`,
+    // it is always a sub-sequence of `col_id_to_def_pos`.
+    for (auto iter = schema_snapshot->getColId2DefPosMap().begin(); iter != schema_snapshot->getColId2DefPosMap().end();
```

Comment:
can we use `for (const auto & [col_id, def_pos]: schema_snapshot->getColId2DefPosMap())` here?

Reply:
Yes, ranged-for and a loop from begin_it to end_it is equivalent.
https://en.cppreference.com/w/cpp/language/range-for

Reply:
fixed